lab-javascript-clue/cluedo.js

```diff
@@ -0,0 +1,72 @@
+
+var suspectList = [mrGreen = {
+  name: "Jacob Green",
+  colour: "green",
+  occupation: "unknown",
+  age: undefined,
+  description: "He has a lot of connections and is always willing to help people out -- for a price."
+}, drOrchid = {name: "Doctor Orchid"}, victor = {name: "Victor Plum"},
+kasandra = {name: "Kasandra Scarlet"}, eleanor = {name: "Eleanor Peacock"},
+jack = {name: "Jack Mustard"}
+];
+
+var weaponList = [ rope = {
+  name: "rope",
+  material: "cotton",
+  color: "dirty white",
+  dimensions: "2 meters",
+  hurtFactor: 1,
+  weight: "light"
+}, knife = {name: "knife"}, candlestick = {name: "candlestick"},
+dumbbell = {name: "dumbbell"}, poison = {name: "poison"},
+axe = {name: "axe"}, bat = {name: "bat"}, trophy = {name: "trophy"},
+pistol = {name: "pistol"}
+];
+
+
+var roomsList = [ hall = {
+  sqrMeters: 20,
+  windows: true,
+  locked: false
+}, diningRoom = {name: "diningRoom"}, conservatory = {name: "conservatory"},
+kitchen = {name: "kitchen"}, study = {name: "study"}, library = {name: "library"},
+billiardRoom = {name: "billiardRoom"}, lounge = {name: "lounge"},
+ballRoom = {name: "ballRoom"}, spa = {name: "spa"}, livingRoom = {name: "livingRoom"},
+observatory = {name: "observatory"}, theatre = {name: "theatre"},
+guestHouse = {name: "guestHouse"}, patio = {name: "patio"}
+];
+
+
+// Method to randomly select one element from a card stack (General)
+function randomCard(cardList) {
+  var min = 0;
+  var max = cardList.length - 1;
+  var randNum = Math.floor(Math.random() * (max - min));
+  var card = cardList[randNum];
+  return card;
+};
+
+// Function that calls the method above for each card stack
+
+var allList = [suspectList, weaponList, roomsList];
+
+function randomList(list) {
+  var result = [];
+  for (var i = 0; i < list.length; i+=1) {
+    result.push(randomCard(list[i]));
```

Comment:
you create the object structure in the best possible way, by giving the same name property to all objects you can automatize your code and make it more efficient. The problem you didn't take full advantage of what you build, instead of accessing the name property in line 69 3 times you can access it directly in the line 56 once